ci/test/04_install.sh

```diff
@@ -27,6 +27,11 @@ export P_CI_DIR="$PWD"
 
 if [ -z "$DANGER_RUN_CI_ON_HOST" ]; then
   echo "Creating $DOCKER_NAME_TAG container to run in"
+  LOCAL_UID=$(id -u)
+  LOCAL_GID=$(id -g)
+
+  # the name isn't important, so long as we use the same UID
+  LOCAL_USER=nonroot
```

Comment:
Sorry, wrong alarm. I think your code is all fine. It is not possible to set `DIR_QA_ASSETS`, because it is not mounted. 

Any folder that is mounted (like `CCACHE_DIR`), will be created before mounting, see the comment `# Create folders that are mounted into the docker`. Also, the `BASE_ROOT_DIR`, which may live outside of `/home/nonroot` in the docker, will be chown'd to `nonroot`.

So the code should be good as-is. (I'll continue testing a bit)